src/plugins/data/public/search/search_interceptor.ts

```diff
@@ -172,34 +173,95 @@ export class SearchInterceptor {
 
     const combinedSignal = getCombinedSignal(signals);
     const cleanup = () => {
-      this.timeoutSubscriptions.remove(subscription);
```

Comment:
Then it seems like `timeoutSubscriptions` are not used anymore and could be removed?